src/components/gltf-model-plus.js

```diff
@@ -187,133 +187,95 @@ function cachedLoadGLTF(src, onProgress) {
   });
 }
 
-AFRAME.registerElement("a-gltf-entity", {
-  prototype: Object.create(AFRAME.AEntity.prototype, {
-    load: {
-      async value() {
-        if (this.hasLoaded || !this.parentEl) {
-          return;
-        }
-
-        // The code above and below this are from AEntity.prototype.load, we need to monkeypatch in gltf loading mid function
-        this.loadTemplates();
-        await this.applySrc(this.getAttribute("src"));
-        //
-
-        AFRAME.ANode.prototype.load.call(this, () => {
-          // Check if entity was detached while it was waiting to load.
-          if (!this.parentEl) {
-            return;
-          }
+AFRAME.registerComponent("gltf-model-plus", {
+  schema: {
+    src: { type: "string" },
+    inflate: { default: false }
+  },
 
-          this.updateComponents();
-          if (this.isScene || this.parentEl.isPlaying) {
-            this.play();
-          }
-        });
-      }
-    },
-
-    loadTemplates: {
-      value() {
-        this.templates = [];
-        this.querySelectorAll(":scope > template").forEach(templateEl =>
-          this.templates.push({
-            selector: templateEl.getAttribute("data-selector"),
-            templateRoot: document.importNode(templateEl.content.firstElementChild, true)
-          })
-        );
-      }
-    },
-
-    applySrc: {
-      async value(src) {
-        try {
-          // If the src attribute is a selector, get the url from the asset item.
-          if (src && src.charAt(0) === "#") {
-            const assetEl = document.getElementById(src.substring(1));
-
-            const fallbackSrc = assetEl.getAttribute("src");
-            const highSrc = assetEl.getAttribute("high-src");
-            const lowSrc = assetEl.getAttribute("low-src");
-
-            if (highSrc && window.APP.quality === "high") {
-              src = highSrc;
-            } else if (lowSrc && window.APP.quality === "low") {
-              src = lowSrc;
-            } else {
-              src = fallbackSrc;
-            }
-          }
+  init() {
+    this.loadTemplates();
+  },
 
-          if (src === this.lastSrc) return;
-          this.lastSrc = src;
+  update() {
+    this.applySrc(this.data.src);
```

Comment:
Add a `.catch` clause here so that any errors will be handled correctly.

Reply:
applySrc checks for errors, so it will always resolve.